Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -7288,6 +7288,30 @@ RefPtr<CSSValue> consumeGridAutoFlow(CSSParserTokenRange& range)
     return parsedValues;
 }
 
+
+RefPtr<CSSValueList> consumeMasonryAutoFlow(CSSParserTokenRange& range)
+{
+    RefPtr<CSSPrimitiveValue> packOrNextValue = consumeIdent<CSSValuePack, CSSValueNext>(range);
+    RefPtr<CSSPrimitiveValue> definiteFirstOrOrderedValue = consumeIdent<CSSValueDefiniteFirst, CSSValueOrdered>(range);
+
+    if (!packOrNextValue) {
+        packOrNextValue = consumeIdent<CSSValuePack, CSSValueNext>(range);
+        if (!packOrNextValue)
+            packOrNextValue = CSSValuePool::singleton().createIdentifierValue(CSSValuePack);
+    }
+
+    RefPtr<CSSValueList> parsedValues = CSSValueList::createSpaceSeparated();
+    if (packOrNextValue) {
+        CSSValueID packOrNextValueID = packOrNextValue->valueID();
+        if (!definiteFirstOrOrderedValue || (definiteFirstOrOrderedValue && definiteFirstOrOrderedValue->valueID() == CSSValueID::CSSValueDefiniteFirst) || packOrNextValueID == CSSValueID::CSSValueNext)
```

Comment:
This condition might be easier to read as the inverse set of checks?

If first value is pack, and second value exists and is not definite-first:

Reply:
I'm finding it very weird that the spec has an explicit 'definite-first' value, which differs from the 'grid-auto-flow' value (where 'sparse' is just an implicit default state, but can't ever be specified).

We should never serialise `definite-first` back though, right? We're aiming for shortest possible serialisation and canonical ordering, so it'll always be omitted.

Reply:
Actually, I also find it weird that both grid-auto-flow and masonry-auto-flow are clearly just two different boolean options that are shoved into a single property. If they really insist on doing that, they could at least be consistent with each other though.

Reply:
Yeah it is definitely strange. In terms of changing the conditional, what wouldn't we want to set the value to pack If the second value is `definite-first`? If the first value is `pack` and the second value is `ordered`, then we do not want to append `packOrNextValue`?